org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/UiSession.java

```diff
@@ -593,9 +594,14 @@ public void dispose() {
     uninstallUiDataAvailableListener();
     signalPoller(); // Notify waiting requests - should not delay web-container shutdown
 
-    m_jsonAdapterRegistry.disposeAdapters();
-    m_httpContext.clear();
-    m_currentJsonResponse = null;
+    // The model may run tasks right now, that alter json adapters, and this may lead to an unreliable state of the JSON layer.
+    // Example: creating and initializing a new json adapter is NOT done atomically and disposing an adapter before it is initialized may lead to errors depending on thy type of json adapter.
+    // Therefore, run this in a model job to ensure a consistent state of the JSON layer before disposing it.
+    runInModelJob(() -> {
+      m_jsonAdapterRegistry.disposeAdapters();
+      m_httpContext.clear();
+      m_currentJsonResponse = null;
+    }, "Dispose UI session", true);
```

Comment:
Jobs names should be more consistent.
- "Dispose UI session" --> "Dispose" or "Disposing"? "UI" or "Ui"?
- "Detaching Gui" --> "Detaching" or "Detach"? "Gui", "GUI" or "UI"?